qwandaq/src/main/java/life/genny/qwandaq/utils/BaseEntityUtils.java

```diff
@@ -182,6 +197,24 @@ public BaseEntity getBaseEntityByCode(String productCode, String code) {
 			try {
 				if (databaseUtils == null) {
 					log.error("databaseUtils is null");
+					// Arc.container().requestContext().activate();
+					Arc.container().instance(DatabaseUtils.class);
+					// databaseUtils = new DatabaseUtils();
+					EntityManagerFactory factory = Persistence.createEntityManagerFactory("genny");
+					entityManager = factory.createEntityManager();
+					// entityManager =
+					// Persistence.createEntityManagerFactory("genny").createEntityManager();
+					if (entityManager == null) {
+						log.error("entityManager is null");
+					}
+					if (databaseUtils == null) {
+						log.error("databaseUtils is still null");
+						databaseUtils = new DatabaseUtils();
+
+						databaseUtils.setEntityManager(entityManager);
```

Comment:
This line is common in both "if" & "else" blocks and could be moved outside after "else"

Reply:
It is because the Arc.container().instance(DatabaseUtils.class) was supposed to reload the databaseUtils....

Reply:
It could be refactored for readability (remove "else"), like:

if (databaseUtils == null) {
	log.error("databaseUtils is still null");
	databaseUtils = new DatabaseUtils();
}
databaseUtils.setEntityManager(entityManager);